pkg/manager/manager.go

```diff
@@ -289,17 +290,82 @@ func (m *manager) AttachRepresentor(conf *types.PluginConf) error {
 		if err = utils.BridgeTrunkVlanAdd(m.nLink, rep, conf.Trunk); err != nil {
 			return fmt.Errorf("failed to add trunk VLAN for representor %s: %v", conf.Representor, err)
 		}
+		log.Info().Msgf("Setting multiple VLANs for rep %s: %v", conf.Representor, conf.Trunk)
 	}
 
 	if conf.Vlan > 0 {
 		if err = utils.BridgePVIDVlanAdd(m.nLink, rep, conf.Vlan); err != nil {
 			return fmt.Errorf("failed to set VLAN for representor %s: %v", conf.Representor, err)
 		}
+		log.Info().Msgf("Setting PVID VLAN for rep %s: %d", conf.Representor, conf.Vlan)
+	}
+
+	// add vlan config to uplink if configured to do so
+	if conf.SetUplinkVlan {
+		if err = m.addUplinkVlans(conf); err != nil {
+			return fmt.Errorf("failed to add vlan to parent uplink %v", err)
+		}
 	}
 
 	return nil
 }
 
+func (m *manager) addUplinkVlans(conf *types.PluginConf) error {
+	var uplink netlink.Link
+	var err error
+
+	uplink, err = m.getPFUplinkOrBond(conf.PFName)
+	if err != nil {
+		return fmt.Errorf("failed to lookup PF for VF-Representor - PFName:%s: VF:%s %v",
+			conf.PFName, conf.Representor, err)
+	}
+
+	var vlans []int
+	if len(conf.Trunk) > 0 {
+		vlans = conf.Trunk
+	}
+
+	if conf.Vlan > 0 {
+		vlans = append(vlans, conf.Vlan)
+	}
+
+	if err = utils.BridgeTrunkVlanAdd(m.nLink, uplink, vlans); err != nil {
+		return fmt.Errorf("failed to add VLANs to parent uplink %s: %v - %v", uplink.Attrs().Name, vlans, err)
+	}
+	log.Info().Msgf("Setting VLANs for uplink %s: %v", uplink.Attrs().Name, vlans)
+
+	return nil
+}
+
+func (m *manager) getPFUplinkOrBond(pfname string) (netlink.Link, error) {
+	var uplink netlink.Link
+	var err error
+	uplink, err = m.nLink.LinkByName(pfname)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to lookup PF %s: %v", pfname, err)
+	}
+
+	if uplink.Attrs().Slave != nil && uplink.Attrs().Slave.SlaveType() == "bond" {
+		var bondLink netlink.Link
+		bondLink, err = m.nLink.LinkByIndex(uplink.Attrs().MasterIndex)
+		if err != nil {
+			return nil, fmt.Errorf("failed to lookup bond interface from slave link master index - Name:%s MasterIndex:%d %v",
+				uplink.Attrs().Name, uplink.Attrs().MasterIndex, err)
+		}
+
+		_, isBond := bondLink.(*netlink.Bond)
+		if !isBond {
+			return nil, fmt.Errorf("master index for link is not a bond - PFName:%s Master Link:%s %v",
+				uplink.Attrs().Name, bondLink.Attrs().Name, err)
+		}
+		uplink = bondLink
+		log.Debug().Msgf("Using bond master as uplink: %s", uplink.Attrs().Name)
+	}
```

Comment:
we should fail with unexpected configuration if PF is a slave but not bond (i.e vrf[1]) WDYT ?


[1] https://github.com/vishvananda/netlink/blob/afa2eb2a66aac1f8f370287f236ba93d4c078dd6/link_linux.go#L1919

Reply:
see above.  i was overthinking this, im going to create a `GetParentBondForLink`